extensions/optaplanner-jsonb/runtime/src/main/java/io/quarkus/optaplanner/jsonb/OptaPlannerJsonbConfigCustomizer.java

```diff
@@ -0,0 +1,47 @@
+package io.quarkus.optaplanner.jsonb;
+
+import javax.inject.Singleton;
+import javax.json.bind.JsonbConfig;
+
+import org.optaplanner.core.api.score.Score;
+import org.optaplanner.persistence.jsonb.api.score.buildin.bendable.BendableScoreJsonbAdapter;
+import org.optaplanner.persistence.jsonb.api.score.buildin.bendablebigdecimal.BendableBigDecimalScoreJsonbAdapter;
+import org.optaplanner.persistence.jsonb.api.score.buildin.bendablelong.BendableLongScoreJsonbAdapter;
+import org.optaplanner.persistence.jsonb.api.score.buildin.hardmediumsoft.HardMediumSoftScoreJsonbAdapter;
+import org.optaplanner.persistence.jsonb.api.score.buildin.hardmediumsoftbigdecimal.HardMediumSoftBigDecimalScoreJsonbAdapter;
+import org.optaplanner.persistence.jsonb.api.score.buildin.hardmediumsoftlong.HardMediumSoftLongScoreJsonbAdapter;
+import org.optaplanner.persistence.jsonb.api.score.buildin.hardsoft.HardSoftScoreJsonbAdapter;
+import org.optaplanner.persistence.jsonb.api.score.buildin.hardsoftbigdecimal.HardSoftBigDecimalScoreJsonbAdapter;
+import org.optaplanner.persistence.jsonb.api.score.buildin.hardsoftdouble.HardSoftDoubleScoreJsonbAdapter;
+import org.optaplanner.persistence.jsonb.api.score.buildin.hardsoftlong.HardSoftLongScoreJsonbAdapter;
+import org.optaplanner.persistence.jsonb.api.score.buildin.simple.SimpleScoreJsonbAdapter;
+import org.optaplanner.persistence.jsonb.api.score.buildin.simplebigdecimal.SimpleBigDecimalScoreJsonbAdapter;
+import org.optaplanner.persistence.jsonb.api.score.buildin.simpledouble.SimpleDoubleScoreJsonbAdapter;
+import org.optaplanner.persistence.jsonb.api.score.buildin.simplelong.SimpleLongScoreJsonbAdapter;
+
+import io.quarkus.jsonb.JsonbConfigCustomizer;
+
+/**
+ * OptaPlanner doesn't use JSON-B, but it does have optional JSON-B support for {@link Score}, etc.
+ */
+@Singleton
+public class OptaPlannerJsonbConfigCustomizer implements JsonbConfigCustomizer {
+
+    @Override
+    public void customize(JsonbConfig config) {
+        config.withAdapters(new BendableScoreJsonbAdapter(),
+                new BendableBigDecimalScoreJsonbAdapter(),
+                new BendableLongScoreJsonbAdapter(),
+                new HardMediumSoftScoreJsonbAdapter(),
+                new HardMediumSoftBigDecimalScoreJsonbAdapter(),
+                new HardMediumSoftLongScoreJsonbAdapter(),
+                new HardSoftScoreJsonbAdapter(),
+                new HardSoftBigDecimalScoreJsonbAdapter(),
+                new HardSoftDoubleScoreJsonbAdapter(),
+                new HardSoftLongScoreJsonbAdapter(),
+                new SimpleScoreJsonbAdapter(),
+                new SimpleBigDecimalScoreJsonbAdapter(),
+                new SimpleDoubleScoreJsonbAdapter(),
+                new SimpleLongScoreJsonbAdapter());
```

Comment:
This exposes the list of score implementations in quarkus, which is a maintenance problem in the long term. We regularly add score implementations upstream.
@cuijulian Can you open a jira to "add OptaPlannerJsonbConfig.getScoreAdaptorList() in optaplanner-persistence-jaxb" and add a comment here, something like "// TODO replace by <jira-url> once released to avoid hardcoding the score implementations" 
Then this PR is mergable. Once that jira is fixed and optaplanner is released, we send in a PR to quarkus to fix the TODO.

Reply:
Jira: https://issues.redhat.com/browse/PLANNER-1922. Comment added, changes pushed in latest commit.